packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -1026,22 +1086,48 @@ class _CupertinoTimerPickerState extends State<CupertinoTimerPicker> {
   Alignment alignCenterLeft;
   Alignment alignCenterRight;
 
+  FixedExtentScrollController hourController;
+  FixedExtentScrollController minuteController;
+  FixedExtentScrollController secondController;
+
   // The currently selected values of the picker.
   int selectedHour;
   int selectedMinute;
   int selectedSecond;
 
+  CupertinoTimerPickerController timerController;
+
   @override
   void initState() {
     super.initState();
 
-    selectedMinute = widget.initialTimerDuration.inMinutes % 60;
+    timerController = widget.controller ?? CupertinoTimerPickerController();
+    timerController.addListener(_desiredDurationChanged);
+
+    // TODO(jslavitz): Remove this check and fully deprecate widget.initialTimerDuration
+    final Duration initialDuration =
+      widget.initialTimerDuration != Duration.zero ? // ignore: deprecated_member_use_from_same_package
+      widget.initialTimerDuration : widget.controller.desiredTimerDuration; // ignore: deprecated_member_use_from_same_package
+
+    selectedMinute = initialDuration.inMinutes % 60;
 
     if (widget.mode != CupertinoTimerPickerMode.ms)
-      selectedHour = widget.initialTimerDuration.inHours;
+      selectedHour = initialDuration.inHours;
 
     if (widget.mode != CupertinoTimerPickerMode.hm)
-      selectedSecond = widget.initialTimerDuration.inSeconds % 60;
+      selectedSecond = initialDuration.inSeconds % 60;
+
+    timerController = widget.controller ?? CupertinoTimerPickerController();
```

Comment:
Hasn't this already been assigned?

Reply:
Unless widget.controller is null. Then we need a default controller to fill it's place as the api relies on a non null controller.

Reply:
I think Matt is asking why is this line both here and on line 1104.

Reply:
:|